### ISSUE TITLE
AMA intake - non-rated | Finish: Include an "unknown" issue category (NODs)

### ISSUE THREAD

**Issue by abbyraskinUSDS** [department-of-veterans-affairs/caseflow#5501]
We'd like to allow users to add additional issues that are not pulled from the disability profile. This ticket is an initial lightweight approach to adding these issues, with further refinement to figure out when in the process we can input additional necessary data points for AMA reporting (e.g., date of claim/decision).

## Acceptance Criteria
- Adds a section the "Finish processing" step 
- Header: "Enter other issue(s) for review" with button that says "Add issue"
- Body: "If the Veteran included any additional issues you cannot find in the list above, please note them below. Otherwise, leave this section blank."
- If a user selects the "Add issue" button, two fields appear:
  - Typeable dropdown populated with compensation categories (see list below) --> the first option is "unknown issue category", then the list is ordered as below
  - Empty freeform description field
  - Save button
- Selecting the "Save" button creates a new empty row (see mock) 
- Issue count below the "Establish appeal" button should include any issues that have been entered in the "other" section
- If there is an issue category selected, an issue description must be entered, and vice-versa

Out of scope:
- This ticket does not cover copy updates to this page (see https://github.com/department-of-veterans-affairs/caseflow/issues/5623)
- Are we sending/storing non-rated or uncategorized issues any differently from rated issues? @shanear 
- The page will look the same, but this ticket is only for Board NODs (appeals)

## Mocks
Empty state
![image](https://user-images.githubusercontent.com/4042181/40313035-8156aa18-5ce2-11e8-811a-035bc7730e5d.png)

After selecting "add issue" button
![image](https://user-images.githubusercontent.com/4042181/40313014-6c07901e-5ce2-11e8-8793-e2a8bbe09696.png)

After selecting "save" button
![image](https://user-images.githubusercontent.com/4042181/40313060-99420fbe-5ce2-11e8-8a1b-49cdb93690a6.png)

## Categories in dropdown
- Unknown issue category
1. Apportionment
2. Incarceration Adjustments
3. Audit Error Worksheet (DFAS)
4. Active Duty Adjustments
5. Drill Pay Adjustments
6. Character of discharge determinations
7. Income/net worth (pension)
8.  Dependent child
a. Adopted
b. Stepchild
c.  Biological
9. Dependency Spouse
a. Common law marriage
b. Inference of marriage
c.       Deemed valid marriage
10.   Military Retired Pay
11.   Contested Claims (other than apportionment)
12.   Lack of Qualifying Service
13. Other non-rated

## Related
#5623

**Comment by abbyraskinUSDS** [comment]
(we can at least get started on the front end)

**Comment by shanear** [comment]
As far as the data model goes, I think there's basically 2 options:

1) Add the necessary fields to `request_issues`. This means adding at lease `issue_category`, and then possibly reusing `description`. This also means that `rating_issue_reference_id` and `rating_issue_profile_date` will be nil on non-rating request issues.

2) Create a new table for non rating request issues. This means we'll make better use of our space, but the relationships could be awkward, since we'll be concatenating lists of issues from two tables for a lot of things.

Not sure the best option. Just wanted to frame the thought.

**Comment by sansb** [comment]
If we went with Option 1 I think it would be better to also have a column like "type" with enum values rating/nonrating. The explicitness there seems like it would be better than inferring it from other nil columns. I can see an argument for either option but option 1 with a rating/nonrating type column feels most correct to me.

**Comment by leikkisa** [comment]
Maybe issue_category could be "rating" for rated issues, and the other category for all non-rated issues.

**Comment by leikkisa** [comment]
Just for documentation, we decided to go with Option 1, and not explicitly record whether an issue is rating or non-rating in a column, but to make a method that shows how to get non-rating issues.

**Comment by leikkisa** [comment]
@abbyraskinUSDS On the first mock-up, there's a label for issue category and issue description, and in the second one it disappears. Is this actually what we want? If not, do we want it to appear above each set of boxes (once per issue), or only once for all issues?

**Comment by abbyraskinUSDS** [comment]
once above each set please @leikkisa, thank you

**Comment by leikkisa** [comment]
Sweet, that's the easiest option

**Comment by leikkisa** [comment]
@abbyraskinUSDS Would it be okay if we just leave the "Add issue" button, and don't have a separate save button? That will be a bit easier, because we won't have to get rid of the "Add issue" button, and then remove and remake a different "Save" button everytime. The way it's working in Redux, the values get saved in the state on any change anyway.
<img width="843" alt="screen shot 2018-06-01 at 4 08 39 pm" src="https://user-images.githubusercontent.com/6589610/40867044-3cefd53e-65b6-11e8-9eb4-4dd4e8d0805c.png">

**Comment by shanear** [comment]
Updating this issue to include not just NOD, but all lanes.

**Comment by leahbannon** [comment]
@leikkisa I like the `Add issue` button. You're saying include that and remove the `Save` buttons? That sounds fine to me.

I'll also try to do some quick ux tests with this before we push to production -- to double check it's immediately clear to people to how to add more than one issue.

**Comment by leahbannon** [comment]
This looks great -- one issue to fix -- this needs a validation fix for the "Enter other issue(s)" section. (Validation acceptance criteria: If there is an issue category selected, an issue description must be entered, and vice-versa.) Currently, if you have one of the rated issues checked, and there are problems with the non-rated issues fields (ex: dropdown selected and no description and vice versa), you are still able to click "Establish EP" and it throws a "Something went wrong" error. 

p.s. Great catch on the save vs. add issue buttons, @leikkisa

**Comment by leikkisa** [comment]
Yes, we actually wanted to ask you and Abby about the UI details, like we
weren't sure if description was required, etc.  I'll ping you two on Slack
because we have questions about the UI for selecting a non-veteran claimant
as well.

**Comment by abbyraskinUSDS** [comment]
Moving this back until we have answers from AMO re: validation

**Comment by leikkisa** [comment]
Adding this information from another (duplicate) ticket that was created, and closing the other ticket:

For non-rating issues, determine if category and description are required. If category is not required, do we want to default it to "Unknown". If data is not filled in, do we want to grey out the "add issue" button. Find out if someone can finish an intake with only non-rating issues?

**Comment by abbyraskinUSDS** [comment]
@leikkisa both fields are mandatory - should I add that to this ticket or start a new one?

**Comment by abbyraskinUSDS** [comment]
Validations will be part of #5991

**Comment by leikkisa** [comment]
@suzchap This file number which you were able to use for the HLR and SC flows should work: 888451290

**Comment by suzchap** [comment]
LGTM
